### PR TITLE
Single product page

### PR DESCRIPTION
**Pull request by tanishk15sharma**
- style: Single product page 
- feat: `useParams`
- feat: select product size
- feat: add to card/add to wishlist
- feat: loading

### CODE REVIEW

**Review by prathmeshjagtap — APPROVED** [review]
Code is Clean 
And It is nice that you have followed all the good conventions

File: src/App.js [hunk 1]
```diff
@@ -21,6 +22,7 @@ function App() {
         <Route path="/wishlist" element={<WishlistPage />} />
         <Route path="/login" element={<UserLogin />} />
         <Route path="/signup" element={<SignUp />} />
+        <Route path="/product/:productId" element={<SingleProduct />} />
         <Route path="/mockapi" element={<Mockman />} />
         <Route path="/*" element={<PageNotFound />} />
       </Routes>
```
Comment (dikshadhote) [inline src/App.js:28]: Routes can be moved to different file.

File: src/backend/db/products.js [hunk 2]
```diff
@@ -11,13 +11,14 @@ export const products = [
...
     discountPrice: 700,
     actualPrice: 900,
     discount: 20,
     inStock: true,
     fastDelivery: false,
     ratings: 4.5,
+    size: "L",
   },
   {
     _id: uuid(),
```
Comment (dikshadhote) [inline src/backend/db/products.js:24]: You can directly push db items from dev and then pull in branch which you are currently using

File: src/pages/SingleProduct/SingleProduct.css [hunk 3]
```diff
@@ -0,0 +1,88 @@
+.product-title {
+  border-bottom: 2px solid rgb(243, 243, 243);
+  margin-bottom: 2rem;
```
Comment (dikshadhote) [inline src/pages/SingleProduct/SingleProduct.css:3]: use color variable

File: src/pages/SingleProduct/SingleProduct.css [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  padding: 10px 20px;
+  font-weight: 700;
+  border: none;
+  cursor: pointer;
+}
+.primary-btn-color {
+  background-color: #e98272;
+  color: white;
+  transition: all 0.2s;
+  border: 1px solid #e98272;
```
Comment (dikshadhote) [inline src/pages/SingleProduct/SingleProduct.css:68]: use color names

**Review by dikshadhote — APPROVED** [review]
Overall the code is neat and clean !!
Good Job !

File: src/App.js [hunk 5]
```diff
@@ -21,6 +22,7 @@ function App() {
         <Route path="/wishlist" element={<WishlistPage />} />
         <Route path="/login" element={<UserLogin />} />
         <Route path="/signup" element={<SignUp />} />
+        <Route path="/product/:productId" element={<SingleProduct />} />
```
Comment (sharathnair9999) [inline src/App.js:25]: you can omit the `/` in the beginning of path of a route. not needed in react router v6

File: src/components/productListing/ProductCard.js [hunk 6]
```diff
@@ -12,10 +12,13 @@ const ProductCard = ({ product }) => {
   const navigate = useNavigate();
   const { cart, setCart } = useCart();
   const { wishList, setWishList } = useWishlist();
+
   return (
     <div className="product-card" key={product.id}>
       <div className="product-img-div">
-        <img src={product.image} alt={`${product.name} image`} />
+        <Link to={`/product/${product._id}`}>
+          <img src={product.image} alt={`${product.name} image`} />
```
Comment (sharathnair9999) [inline src/components/productListing/ProductCard.js:55]: in the `alt` attribute you shouldn't use the work image/picture/photo etc.

File: src/pages/index.js [hunk 7]
```diff
@@ -3,3 +3,4 @@ export { HomePage } from "./HomePage/HomePage";
 export { ProductsPage } from "./ProductsPage/ProductsPage";
 export { WishlistPage } from "./WishlistPage/WishlistPage";
 export { PageNotFound } from "./404/404";
+export { SingleProduct } from "./SingleProduct/SingleProduct";
```
Comment (sharathnair9999) [inline src/pages/index.js:6]: put an index.js file in the pages folder. import the pages there and from there import into outer pages for better readability.

**Review by sharathnair9999 — APPROVED** [review]
suggested trivial changes

File: src/App.js (hunk 1 above)
Comment (tanishk15sharma) [reply src/App.js:28]: sure.

File: src/App.js (hunk 5 above)
Comment (tanishk15sharma) [reply src/App.js:25]: Thanks , I was unaware of this .

File: src/components/productListing/ProductCard.js (hunk 6 above)
Comment (tanishk15sharma) [reply src/components/productListing/ProductCard.js:55]: okay , I will do the changes

File: src/contexts/products-context.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+const productsContext = createContext(null);
+
+const ProductsContextProvider = ({ children }) => {
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        let response = await axios.get("/api/products");
```
Comment (VedanthB) [inline src/contexts/products-context.js:12]: you should move the api calls into a diff folder

**Review by VedanthB — APPROVED** [review]
Good work Tanishk, 

the code is clean and easy to read. Good variable naming. 

You can merge the PR

**Pull request merged**